### PR TITLE
Rise Phpstan level 2, fix some issues

### PR DESCRIPTION
**Pull request by tmotyl**
### Description (*)

This PR rises phpstan level to 2, and fixes some issues along the way.

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#<issue_number>

### Manual testing scenarios (*)

For easier review, I've grouped similar changes per commit.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/SalesRule/Model/Rule.php [hunk 1]
```diff
@@ -415,7 +414,6 @@ public function acquireCoupon($saveNewlyCreated = true, $saveAttemptCount = 10)
         if ($this->getCouponType() == self::COUPON_TYPE_SPECIFIC) {
             return $this->getPrimaryCoupon();
         }
-        /** @var Mage_SalesRule_Model_Coupon $coupon */
```
Comment (kiatng) [inline app/code/core/Mage/SalesRule/Model/Rule.php:418]: Why delete this line?

File: app/code/core/Mage/SalesRule/Model/Rule.php [hunk 2]
```diff
@@ -367,7 +367,6 @@ public function getCoupons()
     {
         if ($this->_coupons === null) {
             $collection = Mage::getResourceModel('salesrule/coupon_collection');
-            /** @var Mage_SalesRule_Model_Resource_Coupon_Collection */
```
Comment (kiatng) [inline app/code/core/Mage/SalesRule/Model/Rule.php:370]: Can this be moved one line above?

File: app/code/core/Mage/SalesRule/Model/Observer.php [hunk 3]
```diff
@@ -120,7 +120,6 @@ public function sales_order_afterPlace($observer)
                 }
             }
             $coupon = Mage::getModel('salesrule/coupon');
-            /** @var Mage_SalesRule_Model_Coupon */
```
Comment (kiatng) [inline app/code/core/Mage/SalesRule/Model/Observer.php:123]: Can this be moved one line above.

File: app/code/core/Mage/SalesRule/Model/Rule.php (hunk 2 above)
Comment (tmotyl) [reply app/code/core/Mage/SalesRule/Model/Rule.php:370]: This comment is not needed. Both phpstan and phpstorm knows type of this variable.

File: app/code/core/Mage/SalesRule/Model/Observer.php (hunk 3 above)
Comment (tmotyl) [reply app/code/core/Mage/SalesRule/Model/Observer.php:123]: This comment is not needed. Both phpstan and phpstorm know type of this variable.

File: app/code/core/Mage/SalesRule/Model/Rule.php (hunk 1 above)
Comment (tmotyl) [reply app/code/core/Mage/SalesRule/Model/Rule.php:418]: This comment is not needed. Both phpstan and phpstorm knows type of this variable.

File: app/code/core/Mage/SalesRule/Model/Rule.php (hunk 2 above)
Comment (tmotyl) [reply app/code/core/Mage/SalesRule/Model/Rule.php:370]: This comment is not needed. Both phpstan and phpstorm knows type of this variable.

File: app/code/core/Mage/SalesRule/Model/Observer.php (hunk 3 above)
Comment (tmotyl) [reply app/code/core/Mage/SalesRule/Model/Observer.php:123]: This comment is not needed. Both phpstan and phpstorm knows type of this variable.

**Review by ADDISON74 — APPROVED** [review]

**Review by fballiano — APPROVED** [review]

**Comment by sreichel** [comment]
@tmotyl thanks for working on this :)

**Comment by sreichel** [comment]
Raised to level 2 and left 7k lines of ignored errors ....

... here we go :P

(Thanks!)

**Comment by sreichel** [comment]
With next commits the baseline file is "only" around 3k lines ... "only" 600 more errors to check ... :P

**Comment by tmotyl** [comment]
@sreichel sounds like its time to go for level 3 or 4

**Comment by sreichel** [comment]
@tmotyl no, not really. Still 3k lines ... I'd like to recude this to below 11 before change to level 3.

I am thankful for any help :)

**Pull request merged**